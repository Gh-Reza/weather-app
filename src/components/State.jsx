import PropTypes from 'prop-types';

const State = ({
  province, icon, condition, temp,
}) => (
  <div className="d-flex justify-content-between align-items-center p-4 bg-dark text-white mb-3">
    <div>
      <h3>{province}</h3>
    </div>
    <div className="d-flex align-items-center">
      <h3>
        {temp}
        <span>{'\u00b0'}</span>
      </h3>
      <div className="ms-5 text-center">
        <img src={icon} alt="" />
        <p>{condition}</p>
      </div>
    </div>
  </div>
);

export default State;

State.propTypes = {
  province: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  condition: PropTypes.string.isRequired,
  temp: PropTypes.number.isRequired,
};
