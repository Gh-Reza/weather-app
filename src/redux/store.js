import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import countriesReducer from './countries/countriesSlice';

const store = configureStore({
  reducer: {
    countries: countriesReducer,
  },
}, applyMiddleware);

export default store;
