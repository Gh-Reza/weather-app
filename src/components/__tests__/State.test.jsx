import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/store';
import '@testing-library/jest-dom/extend-expect';
import State from '../State';

describe('Tests the Header component', () => {
  const component = (
    <BrowserRouter>
      <Provider store={store}>
        <State
          key={1}
          province="Kabul"
          icon=""
          condition=""
          temp={12}
        />
      </Provider>
    </BrowserRouter>
  );

  test('should match the snapshot', () => {
    const tree = render(component);
    expect(tree).toMatchSnapshot();
  });
});
