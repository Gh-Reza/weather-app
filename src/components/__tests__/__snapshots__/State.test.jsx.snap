// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the Header component should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex justify-content-between align-items-center p-4 bg-dark text-white mb-3"
      >
        <div>
          <h3>
            Kabul
          </h3>
        </div>
        <div
          class="d-flex align-items-center"
        >
          <h3>
            12
            <span>
              °
            </span>
          </h3>
          <div
            class="ms-5 text-center"
          >
            <img
              alt=""
              src=""
            />
            <p />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex justify-content-between align-items-center p-4 bg-dark text-white mb-3"
    >
      <div>
        <h3>
          Kabul
        </h3>
      </div>
      <div
        class="d-flex align-items-center"
      >
        <h3>
          12
          <span>
            °
          </span>
        </h3>
        <div
          class="ms-5 text-center"
        >
          <img
            alt=""
            src=""
          />
          <p />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
